Sources/AngelScript/add_on/scriptbuilder/scriptbuilder.h

```diff
@@ -1,193 +1,216 @@
+// then the function should return a negative value to abort the compilation.
+typedef int (*INCLUDECALLBACK_t)(const char *include, const char *from, CScriptBuilder *builder, void *userParam);
+
+// This callback will be called for each #pragma directive encountered by the builder.
+// The application can interpret the pragmaText and decide what do to based on that.
+// If the callback returns a negative value the builder will report an error and abort the compilation.
+typedef int(*PRAGMACALLBACK_t)(const std::string &pragmaText, CScriptBuilder &builder, void *userParam);
+
+// Helper class for loading and pre-processing script files to
+// support include directives and metadata declarations
+class CScriptBuilder
+{
+public:
+	CScriptBuilder();
+
+	// Start a new module
+	int StartNewModule(asIScriptEngine *engine, const char *moduleName);
+
+	// Load a script section from a file on disk
+	// Returns  1 if the file was included
+	//          0 if the file had already been included before
+	//         <0 on error
+	int AddSectionFromFile(const char *filename);
+
+	// Load a script section from memory
+	// Returns  1 if the section was included
+	//          0 if a section with the same name had already been included before
+	//         <0 on error
+	int AddSectionFromMemory(const char *sectionName,
+							 const char *scriptCode,
+							 unsigned int scriptLength = 0,
+							 int lineOffset = 0);
+
+	// Build the added script sections
+	int BuildModule();
+
+	// Returns the engine
+	asIScriptEngine *GetEngine();
+
+	// Returns the current module
+	asIScriptModule *GetModule();
+
+	// Register the callback for resolving include directive
+	void SetIncludeCallback(INCLUDECALLBACK_t callback, void *userParam);
+
+	// Register the callback for resolving pragma directive
+	void SetPragmaCallback(PRAGMACALLBACK_t callback, void *userParam);
+
+	// Add a pre-processor define for conditional compilation
+	void DefineWord(const char *word);
+
+	// Enumerate included script sections
+	unsigned int GetSectionCount() const;
+	std::string  GetSectionName(unsigned int idx) const;
+
+#if AS_PROCESS_METADATA == 1
+	// Get metadata declared for classes, interfaces, and enums
+	std::vector<std::string> GetMetadataForType(int typeId);
+
+	// Get metadata declared for functions
+	std::vector<std::string> GetMetadataForFunc(asIScriptFunction *func);
+
+	// Get metadata declared for global variables
+	std::vector<std::string> GetMetadataForVar(int varIdx);
+
+	// Get metadata declared for class variables
+	std::vector<std::string> GetMetadataForTypeProperty(int typeId, int varIdx);
+
+	// Get metadata declared for class methods
+	std::vector<std::string> GetMetadataForTypeMethod(int typeId, asIScriptFunction *method);
+#endif
+
+protected:
+	void ClearAll();
+	int  Build();
+	int  ProcessScriptSection(const char *script, unsigned int length, const char *sectionname, int lineOffset);
+	int  LoadScriptSection(const char *filename);
```

Comment:
```suggestion
	virtual int  LoadScriptSection(const char *filename);
```
It's an OpenSpades-specific patch I vaguely remember making in d2debd19fe8c20ea8c419f1143ed77f8e09b57be. This is why `spades::ScriptBuilder::LoadScriptSection` had `override`.